### PR TITLE
refactor: changes build from webpack to vite

### PR DESCRIPTION
**Pull request by andy-brown**
# Details
Experimental spike - putting here to record my progress.  Basically there though.

TLDR: it works, as far as I can tell.  Removing flow from the codebase would make things a lot easier.

## Process
* remove all webpack
* `yarn add vite`
* move `index.html` from examples folder to root, fix up links
* extract the `<script>` from `index.html` into a `main.js` that imports `storyplayer`  (this may not be necessary, but seemed to be in order to get storyplayer in in the correct way to be able to be used by `vite`)
* create the `vite.config.js` file to build, test in dev then adjust to build as a library
	* set the right target in the config (e.g., `build.target: 'es2015'`)
	* `build.lib.formats` to `['cjs']` so it outputs usual `storyplayer.js` file in `/dist`
* adjust `package.json` so that the build scripts use `vite`
* test with `yarn dev` - should bring up local site with demo experiences
* test with `yarn build` - should populate the `/dist` folder
* can then do local link from SPH to SP and test it works as a library

### Flow
Not directly supported.  Installed  `vite-plugin-flow` as a plugin which allowed the code to start in `dev` mode.  However, running the code caused console errors on button handling.  Turns out that flow statements defining class variabless, such as
	 ` _handleButtonClick: Function;`
followed by 
	`this._handleButtonClick = this._handleButtonClick.bind(this);`
would fail to bind.
**fix:** Removing the class variable definition lines fixed things.

**TODO** Removing flow altogether would be sensible

### Error with StoryReasoner (EventEmitter)
	`Uncaught TypeError: Class extends value #<Object> is not a constructor or null`
According to [this post](https://stackoverflow.com/questions/70950466/uncaught-typeerror-class-extends-value-object-is-not-a-constructor-or-null) this is to do with the `events` module not being available in the browser.  The suggested solution `yarn add -S events` seemed to work.

### Static files
A bit tricky - needed to be uncompiled for use in SPH, but there is little config available in `vite` for this [see docs](https://vitejs.dev/guide/assets.html)

Solution:
	* put all image files needed into their own `images` folder inside a new `public` folder inside `src/assets`
	* set this `public` folder as `publicDir` in the `vite.config.js` file - this means that everything in there gets dumped in the root, so we end up with a `dist/images` folder as before.
	* also need to adjust the paths to all these public images in the rest of the code to be just `/images/[filename]`

As a result of the latter, needed to squash eslint warnings in `.eslintrc` `rules`:
```
"import/no-unresolved": 0,
"import/no-absolute-path": 0,
```

Then found that the absolute paths didn't work inside SPH, so reverted all the above...

Now have a `src/assets/public/images` folder which is the `publicDir` in vite config.  This has the `black.png` file that SPH uses as a default splash image.  Can't spot any other images used from here by SPH directly...

### require statements
Needed to replace require statements with import statements for threejs and uuid.  Simple enough!

## Todo
* Still got `babel-eslint` in there - used by `eslint`, so can probably stay
* Working out how to run dev on SP within dev on SPH - not sure if this is possible?

### CODE REVIEW

**Comment by andy-brown** [comment]
Superceded by #712
